src/js/components/ServiceList.js

```diff
@@ -35,8 +36,19 @@ let ServiceList = React.createClass({
     return !_.isEqual(this.props, nextProps) || changedState;
   },
 
-  handleServiceClick: function (serviceName) {
-    this.context.router.transitionTo('dashboard-panel', {serviceName});
+  handleServiceClick: function (service, event) {
+    // Open service in new window/tab if service has a web URL
+    if (service.webui_url &&
```

Comment:
unfortunately, this might not be using the service struct.

but i wonder if it should be